### ISSUE TITLE
Include version number of C#

### ISSUE THREAD

**Issue by kirancobe** [dotnet/docs#36154]
[Enter feedback here]
Please include version number of c# where async main works. My understanding is that its C# 7.1 and above?

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 07164632-3cf8-864e-cd1e-485b73ed95a0
* Version Independent ID: 67bfb75a-8dd8-ecf3-ce15-2a531833e8b5
* Content: [await operator - asynchronously wait for a task to complete ](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/await)
* Content Source: [docs/csharp/language-reference/operators/await.md](https://github.com/dotnet/docs/blob/main/docs/csharp/language-reference/operators/await.md)
* Product: **dotnet-csharp**
* Technology: **csharp-language-reference**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

**Comment by BillWagner** [comment]
Hi @kirancobe 

I'd like to know why you need this information for a version that old. 

We do call out new features in the last three versions, plus the current preview version. Currently, that's C# 9 - C# 11, and any preview C# 12 features available in a public preview.

We've found that adding  this information for all versions adds too much complexity for developers new or somewhat new to C#.

**Comment by kirancobe** [comment]
As per my understanding keyword "async" was not allowed to be added to "Main()" method until C 7.1 and later.
This is a specific case and mentioning version number may be useful in this case.
https://stackoverflow.com/questions/9208921/cant-specify-the-async-modifier-on-the-main-method-of-a-console-app

**Comment by BillWagner** [comment]
@kirancobe 

You're understanding is correct. What I want to drill into is *how far back should we mention versions?*  C# 7.1 was [6 years ago](https://learn.microsoft.com/dotnet/csharp/whats-new/csharp-version-history#c-version-71). That page gives a brief history of when each feature was added to the language.  The SO question was raised 11 years ago, with good answers both for the current behavior, and workarounds for developers still on very old versions.

As I said in my first comment, we've chosen 3 releases as a reasonable timeframe. How far back should it go, and why?

**Comment by BillWagner** [comment]
> 

Correct. That's because `async Main` is supported in the last three versions (actually the last 4+). We don't add version information for features more than 3 versions old.

I'll close this as not planned.
